File: src/Palantiri/SensorManager.cs
using System.Collections.Generic;
using Palantiri.Sensors;

namespace Palantiri
{
	public class SensorManager
	{
		protected static readonly List< Sensor > Sensors = new List< Sensor >();
		protected static SensorManager _sensorManager;

		public static SensorManager GetSensorManager()
		{
			return _sensorManager ?? ( _sensorManager = new SensorManager() );
		}

		protected SensorManager()
		{
		}
	}
}

File: src/Palantiri.Core/SensorManager.fs
module SensorManager
open SensorObservable
open System.Collections.Generic
open System.Collections
open System.Linq
open Serilog

type SensorManager private () =
    static let instance = SensorManager()
    static let Sensors = List< Sensor >()
    static member GetSingleton = instance
    member this.AddSensors sensors = Sensors.AddRange<|sensors
    member this.GetSensorsCount sensors = Sensors.Count
    member this.GetSensorTask sensors = 
        try
            Some<|Sensors.First()
        with
            | _ as ex -> Log.Error(ex, "There is no sensor"); None
